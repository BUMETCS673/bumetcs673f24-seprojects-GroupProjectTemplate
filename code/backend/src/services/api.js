import axios from 'axios';

const API_URL = 'http://localhost:8080/api/'; // 替换为你的后端API URL

const api = axios.create({
  baseURL: API_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});

// 请求拦截器，用于添加token到请求头
api.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem('token');
    if (token) {
      config.headers['Authorization'] = 'Bearer ' + token;
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

export default {
  register(user) {
    return api.post('user/register', user);
  },
  login(accountNumber, password) {
    return api.post('user/login', { accountNumber, password });
  },
  forgotPassword(accountNumber) {
    return api.post('user/forgot-password', { accountNumber });
  },
  resetPassword(token, newPassword) {
    return api.post('user/reset-password', { token, newPassword });
  },
  getUserProfile() {
    return api.get('user/profile');
  },
  updateUserProfile(user) {
    return api.put('user/profile', user);
  },
};
