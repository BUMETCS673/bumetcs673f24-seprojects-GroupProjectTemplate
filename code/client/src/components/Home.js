import '../App.css';
import { Line } from 'react-chartjs-2';
import { Chart as ChartJS, CategoryScale, LinearScale, PointElement, LineElement, Title, Tooltip, Legend } from 'chart.js';
import { Typography } from "@mui/material";
import { title, dashboardLineChartContainer } from "./style/styles.js";
import { authenticated } from "../utils/authenticate.js";

// Register the required components for Chart.js
ChartJS.register(CategoryScale, LinearScale, PointElement, LineElement, Title, Tooltip, Legend);


function Home() {
    authenticated();

    //update these constants with data from database
    const dayLabels = ['Day1', 'Day2', 'Day3', 'Day4', 'Day5', 'Day6', 'Day7']

    const weightData = [150, 152, 149, 151, 148, 147, 150];
    const weightGoal = 150;

    const stepsData = [6000, 8000, 7500, 9000, 6500, 5000, 7000];
    const stepsGoal = 10000;

    const sleepData = [6, 7, 5, 8, 6, 7, 5];
    const sleepGoal = 7;

    const waterData = [8, 6, 7, 5, 8, 7, 6];
    const waterGoal = 8;

    const exerciseData = [30, 45, 60, 20, 40, 30, 50];
    const exerciseGoal = 40;

    // Fill in the chart data
    const weightChart = {
        labels: dayLabels, // X-axis labels
        datasets: [
            {
                label: 'Weight in lbs',
                data: weightData, // Y-axis values
                borderColor: 'rgba(75,192,192,1)', // Teal
                pointRadius: 4,
            },
            {
                label: 'Goal',
                data: Array(7).fill(weightGoal), // plot the goal if applicable
                borderColor: 'rgba(75,192,192,.25)', // Teal with 25% transparency
                pointRadius: 0,
            },
        ],
    };
    
    const stepsChart = {
        labels: dayLabels, // X-axis labels
        datasets: [
            {
                label: 'Steps',
                data: stepsData, // Y-axis values
                borderColor: 'rgba(255,99,132,1)', // Red
                pointRadius: 4,
            },
            {
                label: 'Goal',
                data: Array(7).fill(stepsGoal), // plot the goal if applicable
                borderColor: 'rgba(255,99,132,.25)', // Red with 25% transparency
                pointRadius: 0,
            },
        ],
    };
    
    const sleepChart = {
        labels: dayLabels, // X-axis labels
        datasets: [
            {
                label: 'Sleep in hours',
                data: sleepData, // Y-axis values
                borderColor: 'rgba(54,162,235,1)', // Blue
                pointRadius: 4,
            },
            {
                label: 'Goal',
                data: Array(7).fill(sleepGoal), // plot the goal if applicable
                borderColor: 'rgba(54,162,235,.25)', // Blue with 25% transparency
                pointRadius: 0,
            },
        ],
    };
    
    const waterChart = {
        labels: dayLabels, // X-axis labels
        datasets: [
            {
                label: 'Water in glasses',
                data: waterData, // Y-axis values
                borderColor: 'rgba(255,205,86,1)', // Yellow
                pointRadius: 4,
            },
            {
                label: 'Goal',
                data: Array(7).fill(waterGoal), // plot the goal if applicable
                borderColor: 'rgba(255,205,86,.25)', // Yellow with 25% transparency
                pointRadius: 0,
            },
        ],
    };
    
    const exerciseChart = {
        labels: dayLabels, // X-axis labels
        datasets: [
            {
                label: 'Exercise in minutes',
                data: exerciseData, // Y-axis values
                borderColor: 'rgba(153,102,255,1)', // Purple
                pointRadius: 4,
            },
            {
                label: 'Goal',
                data: Array(7).fill(exerciseGoal), // plot the goal if applicable
                borderColor: 'rgba(153,102,255,.25)', // Purple with 25% transparency
                pointRadius: 0,
            },
        ],
    };
    
    
    // Set chart options
    const options = {
        maintainAspectRatio: false,
        scales: {
            y: {
                min: 0
            },
        },
    };

    return (
        <div>
            <Typography variant="h6" gutterBottom sx={title}>
                See your progress:
            </Typography>
            <div style={dashboardLineChartContainer}>
                <Line data={weightChart} options={options} />
            </div>
            <div style={dashboardLineChartContainer}>
                <Line data={stepsChart} options={options} />
            </div>
            <div style={dashboardLineChartContainer}>
                <Line data={sleepChart} options={options} />
            </div>
            <div style={dashboardLineChartContainer}>
                <Line data={waterChart} options={options} />
            </div>
            <div style={dashboardLineChartContainer}>
                <Line data={exerciseChart} options={options} />
            </div>
        </div>
    )
}

export default Home;