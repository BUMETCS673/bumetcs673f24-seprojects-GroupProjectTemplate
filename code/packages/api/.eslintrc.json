{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2020
  },
  "extends": ["prettier", "plugin:node/recommended", "eslint:recommended"],
  "plugins": ["prettier/recommended", "@typescript-eslint/recommended"],
  "rules": {
    "prettier/prettier": "error",
    "@typescrit-eslint/no-unused-vars": "warn",
    "no-console": "off",
    "func-names": "off",
    "no-process-exit": "off",
    "object-shorthand": "off",
    "class-methods-use-this": "off"
  }
}
