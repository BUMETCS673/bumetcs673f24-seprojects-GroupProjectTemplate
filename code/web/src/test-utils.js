import { jsx as _jsx } from "react/jsx-runtime";
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
const AllTheProviders = ({ children }) => {
    return (_jsx(Router, { children: children }));
};
const customRender = (ui, options) => render(ui, Object.assign({ wrapper: AllTheProviders }, options));
export * from '@testing-library/react';
export { customRender as render };
