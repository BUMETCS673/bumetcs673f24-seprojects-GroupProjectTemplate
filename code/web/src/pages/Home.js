var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
// src/pages/Home.tsx
import { useState, useEffect } from 'react';
import Footer from '../components/Footer';
import { Link, useNavigate } from 'react-router-dom';
import Header from '../components/Header';
import api from '../services/api';
import Hero from "../assets/hero-bg.png";
import Filter from "../assets/filter.png";
import Review from "../assets/review.png";
import Mobile from "../assets/mobile.png";
function Home() {
    const navigate = useNavigate();
    const [restaurants, setRestaurants] = useState([]);
    const [searchQuery, setSearchQuery] = useState('');
    const [suggestions, setSuggestions] = useState([]);
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState('');
    useEffect(() => {
        fetchRestaurants();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    const fetchRestaurants = () => __awaiter(this, void 0, void 0, function* () {
        setLoading(true);
        setError('');
        try {
            const response = yield api.get('/restaurants/get-restaurants');
            setRestaurants(response.data);
        }
        catch (err) {
            setError('Failed to fetch restaurants. Please try again later.');
        }
        finally {
            setLoading(false);
        }
    });
    const handleSearchChange = (e) => {
        const query = e.target.value;
        setSearchQuery(query);
        if (query.trim() === '') {
            setSuggestions([]);
        }
        else {
            const filtered = restaurants.filter((restaurant) => restaurant.name.toLowerCase().includes(query.toLowerCase()));
            setSuggestions(filtered.slice(0, 5)); // Limit suggestions to top 5
        }
    };
    const handleSuggestionClick = (restaurant_id) => {
        navigate(`/restaurant/${restaurant_id}`);
    };
    return (_jsxs(_Fragment, { children: [_jsx(Header, {}), _jsxs("main", { className: "flex flex-col items-center", children: [_jsx("section", { className: "w-full bg-cover bg-center h-screen", style: { backgroundImage: `url(${Hero})` }, children: _jsxs("div", { className: "flex flex-col items-center justify-center h-full bg-black bg-opacity-50", children: [_jsx("h1", { className: "text-white text-5xl font-bold mb-4", children: "Find Your Perfect Meal" }), _jsx("p", { className: "text-white text-xl mb-6", children: "Discover restaurants that cater to your dietary needs" }), _jsxs("div", { className: "relative w-80", children: [_jsx("input", { type: "text", placeholder: "Search for restaurants...", value: searchQuery, onChange: handleSearchChange, className: "p-3 rounded-md w-full" }), searchQuery.trim() !== '' && suggestions.length > 0 && (_jsx("ul", { className: "absolute z-10 bg-white border border-gray-300 w-full mt-1 rounded-md max-h-60 overflow-y-auto", children: suggestions.map((suggestion) => (_jsx("li", { className: "p-2 cursor-pointer hover:bg-gray-100", onClick: () => handleSuggestionClick(suggestion.restaurant_id), children: suggestion.name }, suggestion.restaurant_id))) }))] }), _jsxs("div", { className: "flex mt-6 space-x-4", children: [_jsx(Link, { to: "/restaurants?dietaryPreference=vegan", children: _jsx("button", { className: "bg-green-500 text-white px-4 py-2 rounded", children: "Vegan" }) }), _jsx(Link, { to: "/restaurants?dietaryPreference=gluten_free", children: _jsx("button", { className: "bg-yellow-500 text-white px-4 py-2 rounded", children: "Gluten-Free" }) }), _jsx(Link, { to: "/restaurants?dietaryPreference=vegetarian", children: _jsx("button", { className: "bg-blue-500 text-white px-4 py-2 rounded", children: "Vegetarian" }) })] })] }) }), searchQuery.trim() !== '' && suggestions.length === 0 && !loading && (_jsx("section", { className: "w-full max-w-6xl mx-auto py-8 px-4", children: _jsx("p", { className: "text-center text-gray-700", children: "No restaurants found." }) })), _jsxs("section", { className: "py-16 px-4", children: [_jsx("h2", { className: "text-3xl font-bold text-center mb-12", children: "Why Use MenuMatch?" }), _jsxs("div", { className: "flex flex-wrap justify-center space-x-6", children: [_jsxs("div", { className: "w-64 text-center mb-8", children: [_jsx("img", { src: Filter, alt: "Filter", className: "mx-auto mb-4 w-20" }), _jsx("h3", { className: "text-xl font-semibold", children: "Customized Filtering" }), _jsx("p", { children: "Find restaurants that meet your specific dietary preferences." })] }), _jsxs("div", { className: "w-64 text-center mb-8", children: [_jsx("img", { src: Review, alt: "Review", className: "mx-auto mb-4  w-20" }), _jsx("h3", { className: "text-xl font-semibold", children: "Curated Reviews" }), _jsx("p", { children: "Read reviews from users with similar dietary needs." })] }), _jsxs("div", { className: "w-64 text-center mb-8", children: [_jsx("img", { src: Mobile, alt: "Mobile", className: "mx-auto mb-4  w-20" }), _jsx("h3", { className: "text-xl font-semibold", children: "Accessible Anywhere" }), _jsx("p", { children: "Use our platform on both web and mobile devices." })] })] })] }), _jsxs("section", { className: "bg-gray-100 py-16 px-4", children: [_jsx("h2", { className: "text-3xl font-bold text-center mb-12", children: "What Our Users Say" }), _jsxs("div", { className: "flex flex-wrap justify-center space-x-6", children: [_jsxs("div", { className: "w-80 bg-white p-6 rounded-md shadow-md mb-8", children: [_jsx("p", { className: "italic", children: "\"MenuMatch helped me find amazing vegan restaurants when I moved to Boston.\"" }), _jsx("h4", { className: "mt-4 font-semibold", children: "- Alex, Vegan" })] }), _jsxs("div", { className: "w-80 bg-white p-6 rounded-md shadow-md mb-8", children: [_jsx("p", { className: "italic", children: "\"As someone who eats vegetarian, this platform is a lifesaver.\"" }), _jsx("h4", { className: "mt-4 font-semibold", children: "- Fatima, Vegetarian" })] }), _jsxs("div", { className: "w-80 bg-white p-6 rounded-md shadow-md mb-8", children: [_jsx("p", { className: "italic", children: "\"Finding gluten-free options has never been easier!\"" }), _jsx("h4", { className: "mt-4 font-semibold", children: "- John, Gluten-Free" })] })] })] }), _jsxs("section", { className: "py-16 px-4 text-center", children: [_jsx("h2", { className: "text-3xl font-bold mb-6", children: "Join MenuMatch Today" }), _jsx("p", { className: "mb-8", children: "Sign up to start discovering restaurants that fit your dietary needs." }), _jsx(Link, { to: "/login", children: _jsx("button", { className: "bg-red-500 text-white px-6 py-3 rounded-md text-lg", children: "Get Started" }) })] })] }), _jsx(Footer, {})] }));
}
export default Home;
