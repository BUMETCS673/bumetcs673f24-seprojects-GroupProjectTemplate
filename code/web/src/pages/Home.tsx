// src/pages/Home.tsx

import React, { useState, useEffect } from 'react';
import Footer from '../components/Footer';
import { Link, useNavigate } from 'react-router-dom';
import Header from '../components/Header';
import RestaurantCard from '../components/RestaurantCard';
import api from '../services/api';
import Hero from "../assets/hero-bg.png"
import Filter from "../assets/filter.png"
import Review from "../assets/review.png"
import Mobile from "../assets/mobile.png"

interface Restaurant {
    restaurant_id: number;
    name: string;
    address: string;
    phone: string;
    overall_rating: number;
    vegan_rating: number;
    gluten_free_rating: number;
    vegetarian_rating: number;
    // Add other fields if necessary
}

function Home() {
    const navigate = useNavigate();

    const [restaurants, setRestaurants] = useState<Restaurant[]>([]);
    const [searchQuery, setSearchQuery] = useState<string>('');
    const [suggestions, setSuggestions] = useState<Restaurant[]>([]);
    const [loading, setLoading] = useState<boolean>(true);
    const [error, setError] = useState<string>('');

    useEffect(() => {
        fetchRestaurants();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const fetchRestaurants = async () => {
        setLoading(true);
        setError('');
        try {
            const response = await api.get('/restaurants/get-restaurants');
            setRestaurants(response.data);
        } catch (err) {
            setError('Failed to fetch restaurants. Please try again later.');
        } finally {
            setLoading(false);
        }
    };

    const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const query = e.target.value;
        setSearchQuery(query);

        if (query.trim() === '') {
            setSuggestions([]);
        } else {
            const filtered = restaurants.filter((restaurant) =>
                restaurant.name.toLowerCase().includes(query.toLowerCase())
            );
            setSuggestions(filtered.slice(0, 5)); // Limit suggestions to top 5
        }
    };

    const handleSuggestionClick = (restaurant_id: number) => {
        navigate(`/restaurant/${restaurant_id}`);
    };

    return (
        <>
            <Header />
            <main className="flex flex-col items-center">
                {/* Hero Section */}
                <section
                    className="w-full bg-cover bg-center h-screen"
                    style={{ backgroundImage: `url(${Hero})` }}
                >
                    <div className="flex flex-col items-center justify-center h-full bg-black bg-opacity-50">
                        <h1 className="text-white text-5xl font-bold mb-4">Find Your Perfect Meal</h1>
                        <p className="text-white text-xl mb-6">
                            Discover restaurants that cater to your dietary needs
                        </p>
                        {/* Search Bar */}
                        <div className="relative w-80">
                            <input
                                type="text"
                                placeholder="Search for restaurants..."
                                value={searchQuery}
                                onChange={handleSearchChange}
                                className="p-3 rounded-md w-full"
                            />
                            {searchQuery.trim() !== '' && suggestions.length > 0 && (
                                <ul className="absolute z-10 bg-white border border-gray-300 w-full mt-1 rounded-md max-h-60 overflow-y-auto">
                                    {suggestions.map((suggestion) => (
                                        <li
                                            key={suggestion.restaurant_id}
                                            className="p-2 cursor-pointer hover:bg-gray-100"
                                            onClick={() => handleSuggestionClick(suggestion.restaurant_id)}
                                        >
                                            {suggestion.name}
                                        </li>
                                    ))}
                                </ul>
                            )}
                        </div>
                        {/* Dietary Preferences */}
                        <div className="flex mt-6 space-x-4">
                            <Link to="/restaurants?dietaryPreference=vegan">
                                <button className="bg-green-500 text-white px-4 py-2 rounded">Vegan</button>
                            </Link>
                            <Link to="/restaurants?dietaryPreference=gluten_free">
                                <button className="bg-yellow-500 text-white px-4 py-2 rounded">Gluten-Free</button>
                            </Link>
                            <Link to="/restaurants?dietaryPreference=vegetarian">
                                <button className="bg-blue-500 text-white px-4 py-2 rounded">Vegetarian</button>
                            </Link>
                        </div>
                    </div>
                </section>

                {/* Search Results */}
                {searchQuery.trim() !== '' && suggestions.length === 0 && !loading && (
                    <section className="w-full max-w-6xl mx-auto py-8 px-4">
                        <p className="text-center text-gray-700">No restaurants found.</p>
                    </section>
                )}

                {/* Features Section */}
                <section className="py-16 px-4">
                    <h2 className="text-3xl font-bold text-center mb-12">Why Use MenuMatch?</h2>
                    <div className="flex flex-wrap justify-center space-x-6">
                        {/* Feature 1 */}
                        <div className="w-64 text-center mb-8">
                            <img src={Filter} alt="Filter" className="mx-auto mb-4 w-20" />
                            <h3 className="text-xl font-semibold">Customized Filtering</h3>
                            <p>Find restaurants that meet your specific dietary preferences.</p>
                        </div>
                        {/* Feature 2 */}
                        <div className="w-64 text-center mb-8">
                            <img src={Review} alt="Review" className="mx-auto mb-4  w-20" />
                            <h3 className="text-xl font-semibold">Curated Reviews</h3>
                            <p>Read reviews from users with similar dietary needs.</p>
                        </div>
                        {/* Feature 3 */}
                        <div className="w-64 text-center mb-8">
                            <img src={Mobile} alt="Mobile" className="mx-auto mb-4  w-20" />
                            <h3 className="text-xl font-semibold">Accessible Anywhere</h3>
                            <p>Use our platform on both web and mobile devices.</p>
                        </div>
                    </div>
                </section>

                {/* Testimonials Section */}
                <section className="bg-gray-100 py-16 px-4">
                    <h2 className="text-3xl font-bold text-center mb-12">What Our Users Say</h2>
                    <div className="flex flex-wrap justify-center space-x-6">
                        {/* Testimonial 1 */}
                        <div className="w-80 bg-white p-6 rounded-md shadow-md mb-8">
                            <p className="italic">
                                "MenuMatch helped me find amazing vegan restaurants when I moved to Boston."
                            </p>
                            <h4 className="mt-4 font-semibold">- Alex, Vegan</h4>
                        </div>
                        {/* Testimonial 2 */}
                        <div className="w-80 bg-white p-6 rounded-md shadow-md mb-8">
                            <p className="italic">
                                "As someone who eats vegetarian, this platform is a lifesaver."
                            </p>
                            <h4 className="mt-4 font-semibold">- Fatima, Vegetarian</h4>
                        </div>
                        {/* Testimonial 3 */}
                        <div className="w-80 bg-white p-6 rounded-md shadow-md mb-8">
                            <p className="italic">"Finding gluten-free options has never been easier!"</p>
                            <h4 className="mt-4 font-semibold">- John, Gluten-Free</h4>
                        </div>
                    </div>
                </section>

                {/* Call to Action */}
                <section className="py-16 px-4 text-center">
                    <h2 className="text-3xl font-bold mb-6">Join MenuMatch Today</h2>
                    <p className="mb-8">
                        Sign up to start discovering restaurants that fit your dietary needs.
                    </p>
                    <Link to="/login">
                        <button className="bg-red-500 text-white px-6 py-3 rounded-md text-lg">
                            Get Started
                        </button>
                    </Link>
                </section>
            </main>
            <Footer />
        </>
    );
}

export default Home;
