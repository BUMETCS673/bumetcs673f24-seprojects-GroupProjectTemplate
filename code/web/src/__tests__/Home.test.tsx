import React from 'react';
import { render, screen, fireEvent, waitFor } from '../test-utils'; // Adjust import path as needed
import '@testing-library/jest-dom';
import Home from '../../src/pages/Home';
import api from '../../src/services/api';

// Mock the API response
const mockRestaurants = [
  {
    restaurant_id: 1,
    name: 'Vegan Delights',
    address: '123 Plant St',
    phone: '123-4567',
    overall_rating: 4.5,
    vegan_rating: 4.8,
    gluten_free_rating: 4.2,
    vegetarian_rating: 4.5,
  },
  {
    restaurant_id: 2,
    name: 'Gluten-Free Haven',
    address: '456 Wheat Ln',
    phone: '987-6543',
    overall_rating: 4.0,
    vegan_rating: 3.5,
    gluten_free_rating: 4.9,
    vegetarian_rating: 4.0,
  },
];

// Mocking the API get request
jest.mock('../../src/services/api');
(api.get as jest.Mock).mockResolvedValue({ data: mockRestaurants });

// Mocking useNavigate from react-router-dom
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => {
  const original = jest.requireActual('react-router-dom');
  return {
    ...original,
    useNavigate: () => mockNavigate,
  };
});

describe('Home Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders Header and Footer', async () => {
    render(<Home />);
    // Check header
    expect(screen.getByText(/MenuMatch/i)).toBeInTheDocument();
    // Wait for API call to complete to ensure the component has fully rendered
    await waitFor(() => expect(api.get).toHaveBeenCalledTimes(1));
    // Check footer
    const currentYear = new Date().getFullYear().toString();
    expect(screen.getByText(`Â© ${currentYear} MenuMatch. All rights reserved.`)).toBeInTheDocument();
  });

  it('fetches restaurants and shows hero section', async () => {
    render(<Home />);
    await waitFor(() => expect(api.get).toHaveBeenCalledWith('/restaurants/get-restaurants'));
    // Hero section heading
    expect(screen.getByRole('heading', { level: 1, name: /Find Your Perfect Meal/i })).toBeInTheDocument();
    // Dietary preference buttons
    expect(screen.getByRole('button', { name: /Vegan/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Gluten-Free/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Vegetarian/i })).toBeInTheDocument();
  });

  it('shows suggestions when typing a search query', async () => {
    render(<Home />);
    // Wait for data to be loaded
    await waitFor(() => expect(api.get).toHaveBeenCalledTimes(1));

    const searchInput = screen.getByPlaceholderText(/Search for restaurants/i) as HTMLInputElement;
    fireEvent.change(searchInput, { target: { value: 'vegan' } });

    // Suggestions should appear
    await waitFor(() => {
      expect(screen.getByText('Vegan Delights')).toBeInTheDocument();
    });
  });

  it('navigates to restaurant page when a suggestion is clicked', async () => {
    render(<Home />);
    await waitFor(() => expect(api.get).toHaveBeenCalledTimes(1));

    const searchInput = screen.getByPlaceholderText(/Search for restaurants/i) as HTMLInputElement;
    fireEvent.change(searchInput, { target: { value: 'gluten' } });

    await waitFor(() => {
      expect(screen.getByText('Gluten-Free Haven')).toBeInTheDocument();
    });

    // Click the suggestion
    fireEvent.click(screen.getByText('Gluten-Free Haven'));
    expect(mockNavigate).toHaveBeenCalledWith('/restaurant/2');
  });

  it('displays "No restaurants found." when search yields no suggestions', async () => {
    render(<Home />);
    await waitFor(() => expect(api.get).toHaveBeenCalledTimes(1));

    const searchInput = screen.getByPlaceholderText(/Search for restaurants/i) as HTMLInputElement;
    fireEvent.change(searchInput, { target: { value: 'NonExistentRestaurant' } });

    await waitFor(() => {
      // No suggestions should appear
      expect(screen.queryByRole('list')).not.toBeInTheDocument();
      // "No restaurants found." should show up after no suggestions
      expect(screen.getByText('No restaurants found.')).toBeInTheDocument();
    });
  });

  it('renders features, testimonials, and CTA sections', async () => {
    render(<Home />);
    // Wait for data load
    await waitFor(() => expect(api.get).toHaveBeenCalledTimes(1));

    // Features section
    expect(screen.getByText(/Why Use MenuMatch\?/i)).toBeInTheDocument();
    expect(screen.getByText(/Customized Filtering/i)).toBeInTheDocument();
    expect(screen.getByText(/Curated Reviews/i)).toBeInTheDocument();
    expect(screen.getByText(/Accessible Anywhere/i)).toBeInTheDocument();

    // Testimonials
    expect(screen.getByText(/What Our Users Say/i)).toBeInTheDocument();
    expect(screen.getByText(/"MenuMatch helped me find amazing vegan restaurants/i)).toBeInTheDocument();

    // CTA
    expect(screen.getByText(/Join MenuMatch Today/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Get Started/i })).toBeInTheDocument();
  });
});
