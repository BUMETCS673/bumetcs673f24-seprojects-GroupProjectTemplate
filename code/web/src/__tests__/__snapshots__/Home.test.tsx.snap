// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="bg-white shadow-md fixed top-0 left-0 right-0 z-50"
  >
    <div
      class="container mx-auto flex items-center justify-between p-4"
    >
      <a
        class="text-2xl font-bold text-red-500"
        data-discover="true"
        href="/"
      >
        MenuMatch
      </a>
      <nav
        class="space-x-6"
      >
        <a
          class="text-gray-700 hover:text-red-500"
          data-discover="true"
          href="/"
        >
          Home
        </a>
        <a
          class="text-gray-700 hover:text-red-500"
          data-discover="true"
          href="/restaurants"
        >
          Restaurants
        </a>
        <a
          class="text-gray-700 hover:text-red-500"
          data-discover="true"
          href="/about"
        >
          About Us
        </a>
        <a
          class="text-gray-700 hover:text-red-500"
          data-discover="true"
          href="/contact"
        >
          Contact
        </a>
      </nav>
      <div
        class="space-x-4"
      >
        <a
          class="bg-red-500 text-white px-4 py-2 rounded-md"
          data-discover="true"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </header>
  <main
    class="flex flex-col items-center"
  >
    <section
      class="w-full bg-cover bg-center h-screen"
      style="background-image: url(test-file-stub);"
    >
      <div
        class="flex flex-col items-center justify-center h-full bg-black bg-opacity-50"
      >
        <h1
          class="text-white text-5xl font-bold mb-4"
        >
          Find Your Perfect Meal
        </h1>
        <p
          class="text-white text-xl mb-6"
        >
          Discover restaurants that cater to your dietary needs
        </p>
        <div
          class="relative w-80"
        >
          <input
            class="p-3 rounded-md w-full"
            placeholder="Search for restaurants..."
            type="text"
            value=""
          />
        </div>
        <div
          class="flex mt-6 space-x-4"
        >
          <a
            data-discover="true"
            href="/restaurants?dietaryPreference=vegan"
          >
            <button
              class="bg-green-500 text-white px-4 py-2 rounded"
            >
              Vegan
            </button>
          </a>
          <a
            data-discover="true"
            href="/restaurants?dietaryPreference=gluten_free"
          >
            <button
              class="bg-yellow-500 text-white px-4 py-2 rounded"
            >
              Gluten-Free
            </button>
          </a>
          <a
            data-discover="true"
            href="/restaurants?dietaryPreference=vegetarian"
          >
            <button
              class="bg-blue-500 text-white px-4 py-2 rounded"
            >
              Vegetarian
            </button>
          </a>
        </div>
      </div>
    </section>
    <section
      class="py-16 px-4"
    >
      <h2
        class="text-3xl font-bold text-center mb-12"
      >
        Why Use MenuMatch?
      </h2>
      <div
        class="flex flex-wrap justify-center space-x-6"
      >
        <div
          class="w-64 text-center mb-8"
        >
          <img
            alt="Filter"
            class="mx-auto mb-4 w-20"
            src="test-file-stub"
          />
          <h3
            class="text-xl font-semibold"
          >
            Customized Filtering
          </h3>
          <p>
            Find restaurants that meet your specific dietary preferences.
          </p>
        </div>
        <div
          class="w-64 text-center mb-8"
        >
          <img
            alt="Review"
            class="mx-auto mb-4  w-20"
            src="test-file-stub"
          />
          <h3
            class="text-xl font-semibold"
          >
            Curated Reviews
          </h3>
          <p>
            Read reviews from users with similar dietary needs.
          </p>
        </div>
        <div
          class="w-64 text-center mb-8"
        >
          <img
            alt="Mobile"
            class="mx-auto mb-4  w-20"
            src="test-file-stub"
          />
          <h3
            class="text-xl font-semibold"
          >
            Accessible Anywhere
          </h3>
          <p>
            Use our platform on both web and mobile devices.
          </p>
        </div>
      </div>
    </section>
    <section
      class="bg-gray-100 py-16 px-4"
    >
      <h2
        class="text-3xl font-bold text-center mb-12"
      >
        What Our Users Say
      </h2>
      <div
        class="flex flex-wrap justify-center space-x-6"
      >
        <div
          class="w-80 bg-white p-6 rounded-md shadow-md mb-8"
        >
          <p
            class="italic"
          >
            "MenuMatch helped me find amazing vegan restaurants when I moved to Boston."
          </p>
          <h4
            class="mt-4 font-semibold"
          >
            - Alex, Vegan
          </h4>
        </div>
        <div
          class="w-80 bg-white p-6 rounded-md shadow-md mb-8"
        >
          <p
            class="italic"
          >
            "As someone who eats vegetarian, this platform is a lifesaver."
          </p>
          <h4
            class="mt-4 font-semibold"
          >
            - Fatima, Vegetarian
          </h4>
        </div>
        <div
          class="w-80 bg-white p-6 rounded-md shadow-md mb-8"
        >
          <p
            class="italic"
          >
            "Finding gluten-free options has never been easier!"
          </p>
          <h4
            class="mt-4 font-semibold"
          >
            - John, Gluten-Free
          </h4>
        </div>
      </div>
    </section>
    <section
      class="py-16 px-4 text-center"
    >
      <h2
        class="text-3xl font-bold mb-6"
      >
        Join MenuMatch Today
      </h2>
      <p
        class="mb-8"
      >
        Sign up to start discovering restaurants that fit your dietary needs.
      </p>
      <a
        data-discover="true"
        href="/login"
      >
        <button
          class="bg-red-500 text-white px-6 py-3 rounded-md text-lg"
        >
          Get Started
        </button>
      </a>
    </section>
  </main>
  <footer
    class="bg-gray-800 text-white mt-16"
  >
    <div
      class="container mx-auto py-8 flex flex-col md:flex-row justify-between items-center"
    >
      <a
        class="text-2xl font-bold mb-4 md:mb-0"
        data-discover="true"
        href="/"
      >
        MenuMatch
      </a>
      <div
        class="space-x-6"
      >
        <a
          class="hover:text-red-500"
          data-discover="true"
          href="/privacy"
        >
          Privacy Policy
        </a>
        <a
          class="hover:text-red-500"
          data-discover="true"
          href="/terms"
        >
          Terms of Service
        </a>
      </div>
      <div
        class="space-x-4 mt-4 md:mt-0"
      >
        <a
          aria-label="Facebook"
          class="hover:text-red-500"
          href="#"
        >
          FB
        </a>
        <a
          aria-label="Twitter"
          class="hover:text-red-500"
          href="#"
        >
          TW
        </a>
        <a
          aria-label="Instagram"
          class="hover:text-red-500"
          href="#"
        >
          IG
        </a>
      </div>
    </div>
    <div
      class="text-center py-4 bg-gray-900"
    >
      Â© 2024 MenuMatch. All rights reserved.
    </div>
  </footer>
</DocumentFragment>
`;
